docs/modules/data-connections/pages/data-connections-configuration.adoc

```diff
@@ -381,6 +381,147 @@ OPTIONS (
+              http://www.hazelcast.com/schema/client-config/hazelcast-client-config-5.3.xsd">
+
+            <cluster-name>dev</cluster-name>
+              <network>
+                <cluster-members>
+                  <address>172.17.0.2:5701</address>
+                </cluster-members>
+              </network>
+          </hazelcast-client>
+        ]]>
+      </property>
+    </properties>
+    <shared>true</shared>
+  </data-connection>
+</hazelcast>
+----
+<1> (Required) Exactly one of `client_xml`, `client_yml`, `client_xml_path`, `client_yml_path` must be specified.
+<2> Hazelcast client configuration to connect to a remote cluster. See xref:clients:java.adoc#configuring-java-client[Configuring Java Client].
+External file can be specified using `client_xml_path` property instead of embedded configuration file.
+--
+
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  data-connection:
+    my-remote-hazelcast:
+      type: Hz
+      properties:
+        client_yml: | <1>
+          hazelcast-client: <2>
+            cluster-name: dev
+            network:
+              cluster-members:
+                - 172.17.0.2:5701
+      shared: true
+----
+<1> (Required) Exactly one of `client_xml`, `client_yml`, `client_xml_path`, `client_yml_path` must be specified.
+<2> Hazelcast client configuration to connect to a remote cluster. See xref:clients:java.adoc#configuring-java-client[Configuring Java Client]].
+External file can be specified using `client_yml_path` property instead of embedded configuration file.
+--
+
+Java::
++
+--
+[source,java]
+----
+config
+  .addDataConnectionConfig(
+    new DataConnectionConfig()
+      .setName("my-remote-hazelcast")
+      .setType("Hz")
+      .setProperty(
+              "client_xml", <1>
+              "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + <2>
+              "<hazelcast-client xmlns=\"http://www.hazelcast.com/schema/client-config\"" +
+              "      xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"" +
+              "      xsi:schemaLocation=\"http://www.hazelcast.com/schema/client-config" +
+              "      http://www.hazelcast.com/schema/client-config/hazelcast-client-config-5.3.xsd\">" +
+              "    " +
+              "    <cluster-name>dev</cluster-name>" +
+              "    <network>" +
+              "        <cluster-members>" +
+              "            <address>172.17.0.2:5701</address>" +
+              "        </cluster-members>" +
+              "    </network>" +
+              "</hazelcast-client>")
+      .setShared(true)
+  );
+----
+<1> (Required) Exactly one of `client_xml`, `client_yml`, `client_xml_path`, `client_yml_path` must be specified.
+<2> Hazelcast client configuration to connect to a remote cluster. See xref:clients:java.adoc#configuring-java-client[Configuring Java Client].
+External file can be specified using `client_xml_path` or `client_yml_path` property instead of embedded configuration file.
```

Comment:
```suggestion
You can specify an external file with the `client_xml_path` or `client_yml_path` property instead of using an embedded configuration file.
```